Dockerfile

```diff
@@ -1,151 +1,121 @@
-
-# Setup webpack building base images
-# We always do the webpack build on amd64, cause it's way faster
-FROM amd64-node-base as rpi-node-build
-FROM amd64-node-base as amd64-node-build
-FROM amd64-node-base as armv7hf-node-build
-FROM amd64-node-base as aarch64-node-build
-FROM amd64-node-base as i386-node-build
-FROM balenalib/amd64-node:6-build as i386-nlp-node-build
-##############################################################################
-
-FROM $ARCH-node-build as node-build
+ARG NODE_VERSION=10.18.0
 
+FROM balenalib/$ARCH-alpine-supervisor-base:3.11 as BUILD
 
-WORKDIR /usr/src/app
-
-RUN apt-get update \
-	&& apt-get install -y \
-		g++ \
-		git \
-		libsqlite3-dev \
-		make \
-		python \
-		rsync \
-		curl \
-	&& rm -rf /var/lib/apt/lists/
-
-COPY package.json package-lock.json /usr/src/app/
-
-ARG NPM_VERSION
-# We first ensure that every architecture has an npm version
-# which can do an npm ci, then we perform the ci using this
-# temporary version
-RUN curl -LOJ https://www.npmjs.com/install.sh && \
-	# This is required to avoid a bug in uid-number
-	# https://github.com/npm/uid-number/issues/7
-	npm config set unsafe-perm true && \
-	npm_install="#{NPM_VERSION}" npm_config_prefix=/tmp sh ./install.sh && \
-	JOBS=MAX /tmp/bin/npm ci --no-optional --unsafe-perm
-
-COPY webpack.config.js fix-jsonstream.js hardcode-migrations.js tsconfig.json tsconfig.release.json /usr/src/app/
-COPY src /usr/src/app/src
-COPY test /usr/src/app/test
-COPY typings /usr/src/app/typings
-
-RUN npm run test-nolint \
-	&& npm run build
-
-##############################################################################
-
-# Build nodejs dependencies
-FROM $ARCH-node-base as node-deps
-
-RUN [ "cross-build-start" ]
+ARG ARCH
+ARG NODE_VERSION
+ARG NODE_ARCHIVE="node-no-intl-v${NODE_VERSION}-linux-alpine-${ARCH}.tar.gz"
+ARG S3_BASE="https://resin-packages.s3.amazonaws.com"
+ARG NODE_LOCATION="${S3_BASE}/node/v${NODE_VERSION}/${NODE_ARCHIVE}"
+ARG SUMS="SHASUMS256.txt"
+
+# DO NOT REMOVE THE cross-build-* COMMANDS
+# The following commands are absolutely needed. When we
+# build for ARM architectures, we run this Dockerfile
+# through sed, which uncomments these lines. There were
+# other options for achieving the same setup, but this seems
+# to be the least intrusive. The commands start commented
+# out because the default build for balenaCI is amd64 (and
+# we can't run any sed preprocessing on it there)
+# RUN ["cross-build-start"]
 
 WORKDIR /usr/src/app
 
-RUN apt-get update \
-	&& apt-get install -y \
+RUN apk add --no-cache \
 	g++ \
 	git \
-	libsqlite3-dev \
 	make \
 	python \
-	rsync \
 	curl \
-	&& rm -rf /var/lib/apt/lists/
-
-RUN mkdir -p rootfs-overlay && \
-	ln -s /lib rootfs-overlay/lib64
+	binutils \
+	libgcc \
+	libstdc++ \
+	libuv
 
-COPY package.json package-lock.json /usr/src/app/
+# Install node from baleana's prebuilt cache
+RUN curl -SLO "${NODE_LOCATION}" \
+	&& curl  -SLO "${S3_BASE}/${SUMS}" \
```

Comment:
We should really include the SHAs locally, since they're meant to verify we get the expected file but downloading the SHAs from the same place as the binary only means that they didn't get corrupted in transport and says nothing about them containing what we expect

Reply:
Done.